### PR TITLE
Load inbox if not complete or errored

### PR DESCRIPTION
**Pull request by gabriel**
Chat will load inbox if not completed or errored.

### CODE REVIEW

**Comment by gabriel** [comment]
@keybase/react-hackers

File: shared/actions/chat.js [hunk 1]
```diff
@@ -1054,15 +1054,18 @@ function * _ensureValidSelectedChat (onlyIfNoSelection: boolean, forceSelectOnMo
 
 const followingSelector = (state: TypedState) => state.config.following
 
-let _loadedInboxOnce = false
 function * _loadInboxMaybeOnce (action: LoadInbox): SagaGenerator<any, any> {
-  if (!_loadedInboxOnce || action.payload.force) {
-    _loadedInboxOnce = true
+  console.log('Load inbox? complete:', _inboxComplete, 'error:', _inboxError, 'force:', action.payload.force)
+  if (!_inboxComplete || _inboxError || action.payload.force) {
```
Comment (chrisnojima) [inline shared/actions/chat.js]: we don't need to check inboxError here right, complete captures the entire state of this

File: shared/actions/chat.js [hunk 2]
```diff
@@ -1054,15 +1054,18 @@ function * _ensureValidSelectedChat (onlyIfNoSelection: boolean, forceSelectOnMo
 
 const followingSelector = (state: TypedState) => state.config.following
 
-let _loadedInboxOnce = false
 function * _loadInboxMaybeOnce (action: LoadInbox): SagaGenerator<any, any> {
-  if (!_loadedInboxOnce || action.payload.force) {
-    _loadedInboxOnce = true
+  console.log('Load inbox? complete:', _inboxComplete, 'error:', _inboxError, 'force:', action.payload.force)
```
Comment (chrisnojima) [inline shared/actions/chat.js]: 🔥

File: shared/actions/chat.js (hunk 1 above)
Comment (gabriel) [inline shared/actions/chat.js]: It can complete with or without an error

Comment (chrisnojima) [inline shared/actions/chat.js]: yes but complete is always true in this case right. the error just stores the error

Comment (gabriel) [inline shared/actions/chat.js]: IDK if the service fell over (maybe crashed?) then complete wouldn't get called.
If there was a bug that prevented rpc call, etc etc.

**Comment by gabriel** [comment]
I could remove the complete bool if we don't want to handle that failure case though which is what you're suggesting?

**Comment by chrisnojima** [comment]
Actually I take it back but i think the logic is still weird. Ultimately don't we want you do only do the loadInbox if 
1. We're not in the middle of one
2. If we had a problem
1. If we want to force it

so shouldn't it be more like

```
if (_inboxComplete && (_inboxError || action.payload.force)) {
      yield call(_loadInbox)
    }
```

cause _inboxComplete should be called regardless. If we're worried about the case where finish is never called, then we need some kind of race w/ a timeout where we set complete to true with inboxError as timeout instead.

**Comment by gabriel** [comment]
ok.. I think the right check is:

```
if (!_inboxLoading && (!_inboxLoadedOnce || _inboxError || action.payload.force)) {
```

- We don't load if we are loading
- We load if we haven't ever loaded, there was an error loading or we're forcing it

**Comment by gabriel** [comment]
I set _inboxLoading = false on finished and timeout. I set an _inboxError on timeout so it will reload.

There is a while(!finishedCalled) and break here which is redundant and was a little confusing to me. (The timeout doesn't set finishCalled but does break.) We might want to fix this later?

**Comment by chrisnojima** [comment]
lets just fix that now. either remove the break and set finishCalled (or maybe 'allDone' or something) or use the break and make it a while(true). should be a 3 line change either way

**Comment by gabriel** [comment]
Removed `finishedCalled` and use while(true)

**Comment by chrisnojima** [comment]
let's close this PR for now. i'm currently working on async inbox loading so i'll pull in the code you have here (but i have to shift stuff around since we make multiple calls instead of just one now so this stuff will all get moved around a bit)

File: shared/actions/chat/inbox.js [hunk 3]
```diff
@@ -111,13 +117,14 @@ function * onLoadInbox (): SagaGenerator<any, any> {
...
     } else if (incoming.chatInboxFailed) {
-      console.log('ignoring chatInboxFailed', incoming.chatInboxFailed)
+      console.log('chatInboxFailed', incoming.chatInboxFailed)
       requestIdleCallback(() => {
         incoming.chatInboxFailed.response.result()
       }, {timeout: 100})
 
       yield call(delay, 1)
       const error = incoming.chatInboxFailed.params.error
+      _inboxError = error
```
Comment (chrisnojima) [inline shared/actions/chat/inbox.js]: this doesn't mean the entire thing errored, just one row. it shouldn't be involved in the reloading

**Comment by chrisnojima** [comment]
pulled in your code into: https://github.com/keybase/client/pull/6324

**Comment by gabriel** [comment]
👍

**Pull request closed without merging**